crowd_anki/export/note_sorter.py

```diff
@@ -0,0 +1,39 @@
+from ..config.config_settings import ConfigSettings, NoteSortingMethods
+
+
+class NoteSorter:
+    sorting_definitions = {
+        NoteSortingMethods.NO_SORTING: lambda i: i,
```

Comment:
is this actually valid? i.e. if we return just the object itself, the sort method is still going to sort it somehow right?

Reply:
I believe so, but it does not matter because it is not used. That's what should_sort checks for, and skips the sort process if NO_SORTING is selected. If someone sets the selected sort methods to "guid, none" then this sort method will be used, but since there is already a first it will not matter. It'll already be different than your default before this PR, by their own selection, so there should be no issue. Right? :thinking:

Reply:
well tbh, having none in combination with other things does not really make sense to me. But given that we can have it, the only coherent way to implement it is to not change position of items at all if the "none" is provided. I think, given that python sort implementation is stable, this can be achieved by returning some constant value from the key.

Reply:
👍 I'd go with the string that'd explain the reasoning above though, or at least adding a comment

Reply:
Alright, how'd you like the look of this? :grin: 

```python
class NoteSorter:
    sorting_definitions = {
        # NO_SORTING is a special case which should be ignored by should_sort
        # However it's been kept here for edge cases, and for ease of testing
        NoteSortingMethods.NO_SORTING: lambda i: 1,
        
        NoteSortingMethods.GUID: lambda i: i.anki_object.guid,
        NoteSortingMethods.FLAG: lambda i: i.anki_object.flags,
        NoteSortingMethods.TAG: lambda i: i.anki_object.tags,
        NoteSortingMethods.NOTE_MODEL: lambda i: i.anki_object._model["name"],
        NoteSortingMethods.NOTE_MODEL_ID: lambda i: i.anki_object._model["crowdanki_uuid"],
        NoteSortingMethods.FIELD1: lambda i: i.anki_object.fields[0],
        NoteSortingMethods.FIELD2: lambda i: i.anki_object.fields[1]
    }
```

Reply:
I believe the justification for why having `1` there make sense should be there (the whole thing about stable sorting)

Reply:
```python
class NoteSorter:
    sorting_definitions = {
        # NO_SORTING is a special case which should be ignored by should_sort
        # However it's been kept here for edge cases, and for ease of testing
        # It returns 1 to ensure stable sorting, so the results will be in their previous order
        NoteSortingMethods.NO_SORTING: lambda i: 1,

        NoteSortingMethods.GUID: lambda i: i.anki_object.guid,
        NoteSortingMethods.FLAG: lambda i: i.anki_object.flags,
        NoteSortingMethods.TAG: lambda i: i.anki_object.tags,
        NoteSortingMethods.NOTE_MODEL_NAME: lambda i: i.anki_object._model["name"],
        NoteSortingMethods.NOTE_MODEL_ID: lambda i: i.anki_object._model["crowdanki_uuid"],
        NoteSortingMethods.FIELD1: lambda i: i.anki_object.fields[0],
        NoteSortingMethods.FIELD2: lambda i: i.anki_object.fields[1]
    }
    
```

:thinking: ?